### ISSUE TITLE
Support PubSub (XMPP)

### ISSUE THREAD

**Issue by sjehuda** [feedreader/pluto#48]
Greetings!

In XMPP, using PubSub(1) (XEP-0060), it is possible to store posts as Atom Syndication(2) entries into XMPP(3).

There are several XMPP extensions that apply it, namely XEP-0227 (4) and XEP-0472 (5) which make node `urn:xmpp:microblog:0` as a common choice to manage news for client accounts (e.g. `blog@planetplanet.org`), yet a component account (e.g. `pubsub.planetplanet.org`) may use all of its nodes for this purpose.
```
pubsub.planetplanet.org/
|
|-- updates/
    |-- Sombre
     `-- Varied
|-- releases/
 `-- events/
pubsub service/
|
|-- node/
    |-- item
     `-- item
|-- node/
 `-- node/
```
A Ruby library that might be utilized would be XMPP4R.

I have made a script that realizes viewing PubSub node items as Atom XML feeds over HTTP.

[XEP-0060: Publish-Subscribe](https://xmpp.org/extensions/xep-0060.html)
[RFC 4287: The Atom Syndication Format](https://www.rfc-editor.org/rfc/rfc4287)
[Atom Over XMPP](https://datatracker.ietf.org/meeting/66/materials/slides-66-atompub-1.pdf)
[XEP-0277: Microblogging over XMPP](https://xmpp.org/extensions/xep-0277.html)
[XEP-0472: Pubsub Social Feed](https://xmpp.org/extensions/xep-0472.html)
[XMPP4R](http://xmpp4r.github.io/)
[PubSub Over HTTP](https://codeberg.org/poezio/slixmpp/issues/3526)

**Comment by geraldb** [comment]
@sjehuda   may i ask -  are you running any planets with pluto?

**Comment by sjehuda** [comment]
No. Not yet, but I might.

I program in Python,and so I intend to run a "planet" with a Python software and I also intend add this feature to it.

**Comment by geraldb** [comment]
fyi: the planet pluto machinery is in ruby (NOT python).

**Comment by sjehuda** [comment]
I am aware of it.

This is why I have linked to an [XMPP Ruby Library](http://xmpp4r.github.io/) .

I ask for this feature to be added to Pluto because it is vital.

ATOM/XMPP is definitely better than HTML/HTTP.

---

On another note, I intend to add this feature to:
1) Another "planet" software.
2) A static site generator.

**Comment by geraldb** [comment]
thanks for the update.   if i find time i try to learn more about pubsub / xmpp (subscribe).  for static site generatation - a basic static site generator with ready-to-use templates is built-in. as alternative you can use any sstatic site generator. many use jekell - see planet opensuse and others for how to setup with pluto.

**Comment by sjehuda** [comment]
> if i find time i try to learn more about pubsub / xmpp (subscribe).

This feature is not related to "subscribe"; it only relates to uploading XML data (Atom Syndication Format) to PubSub nodes.

The publishing is done by server (e.g. eJabberd, Openfire, Prosody etc.) and subscription is done by client (e.g. Dergchat, Gajim, Kaidan, Psi, Movim etc.)

Software which upload data to PubSub nodes:
- https://github.com/imattau/atomtopubsub (client)
- https://gitgud.io/sjehuda/slixfeed (client)
- https://modules.prosody.im/mod_pubsub_feeds.html (server)

Results can be tested with Movim:
https://mov.im/community/news.movim.eu
https://mov.im/community/blabla.movim.eu

Note: news.movim.eu and blabla.movim.eu function as PubSub services which enclose nodes within them. Each node represents An tom Syndication Feed.

**Comment by geraldb** [comment]
i see. thanks for the update & comments.    if i understand this works like publishing with a new teemplate / setup.   if yes, have a look at the opensuse and friends setup that use their own site generator (jekyll) that formula might work for you too.

**Comment by sjehuda** [comment]
I am not sure what you mean by template.

All I send to XMPP PubSub node is Atom Syndication Format compliant XML elements, including HTML _enclosed_ inside XML (This is part of the RFC 4287 standard).

Concerning to to Jekyll, I have posted a similar ticket there too. See https://github.com/jekyll/jekyll/issues/9583

**Comment by geraldb** [comment]
> I am not sure what you mean by template.
you can generate (export) your feed items (with or without templates) as you like. for an example again see planet open suse (in action) here -> <https://github.com/opensuse/planet-o-o>  
and the generation script here -> <https://github.com/openSUSE/planet-o-o/blob/master/bin/jekyll_planet.rb>

**Comment by sjehuda** [comment]
@mattr [has wrote](https://github.com/jekyll/jekyll/issues/9583#issuecomment-2127410574) that it can be a plugin.

I might volunteer to work on it too. I might join another friend who codes in Ruby too.

If it is relevant to both Jekyll and Pluto, please inform me of your progress in order to coordinate our efforts.

**Comment by geraldb** [comment]
yes please, write a plugin. nothing here (planed). it's all up-to-you. all the best.

**Comment by sjehuda** [comment]
Alright. I might want to use these projects to accomplish this task:

https://git.singpolyma.net/xmpp-blog-utils

https://github.com/adhearsion/blather
